### DISCUSSION TITLE
Welcome to the DIY Dyno project.

### DISCUSSION THREAD

**Discussion by DeeEmm** [DeeEmm/DIY-Dyno#1] [General]
Thanks for taking a look at the DIY Dyno project.

Some of you might have ended up here from my other project -[DIY Flow Bench](https://github.com/DeeEmm/DIY-Flow-Bench) or perhaps even from the [DIY Dyno Facebook page](https://www.facebook.com/groups/dynodiy/)

Whilst this Repo is fairly new, the project has been brewing for a number of years. I got as far as partially building the physical Dyno but then put the project on the back-burner in favour of other projects like the Flow Bench. 

So the time has now come to pick up the project again and develop it into something functional. The decision for this has come partly from the DIY Flow Bench project which is now at release candidate stage. It has a working shield as well as functional software, which not only means I am free to concentrate on other projects, but it also provides a foundation from which the DIY Dyno project can be developed.

Much of the graphical user interface (GUI) and underlying boilerplate code of the flow bench project can be reused for this project. They both work on similar principles, specifically; take a bunch of sensor readings, use the resultant data to calculate an output and then display the output in the form of a graph to the user via the GUI. So to a degree a large part of the hard work is already done.

The remaining work relates to the sensors and math's aspect of the code. The Dyno uses load cells instead of pressure sensors and we are calculating power and torque rather than volumetric flow. Fortunately the code was written in a way that allows it to be easily adapted. 

The first job is to identify what needs to be done. The very basics of the project relate to reading a load cell and tacho signal so that power and torque can be calculated, but there are other features that form part of most Dyno and data acquisition controllers that need to be considered. The first iteration of the project will most likely be fairly basic, but we need to consider where the project is ultimately headed so that future features are included in the roadmap and provision is included on the shield.

At the moment the timeline is undecided. I think that it would be good to get a very basic version 1 up and running that includes just the bare bones of what is required to map a power / torque graph. This would allow people to do Dyno pulls and record some data. For many this is possibly all that is required. Version 2 can then concentrate on adding other features that users want to see such as brake regulation and additional data logging channels.

So whilst there is no code to try or hardware to build at this stage, if you have an interest in the project, please feel free to join in the conversation. 

/DM

**Comment by black-top** [comment]
It's nice to see this project. I thought about diy dyno controller for a long time and collected information and parts for my dyno for few yeras now. I will try to contribute to the development of the project as much as I can.

**Reply by DeeEmm** [reply]
Thanks, good to see you here. Your help on the Flow Bench project has been invaluable. I'm sure it will be the same here too.

**Comment by ParaFlight** [comment]
Hi, Nice project!!!, was searching exactly for a dyno using esp32, are you still working on it? Cheers!
